pkg/daemon/update.go

```diff
@@ -84,6 +84,11 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 		return fmt.Errorf("%s", msg)
 	}
 
+	// Create direcotry path for flexvolume plugin
+	if err = dn.fileSystemClient.MkdirAll(pathFlexVolumePlugin, DefaultDirectoryPermissions); err != nil {
```

Comment:
I don't think we want this as Go code; it should just be an Ignition config like any others.  Specifically an entry in `directories`; see [the Ignition spec](https://coreos.com/ignition/docs/latest/configuration-v2_2.html). 

IOW you'd want this to be a file in `templates` like how we have the `kubelet.yaml` there.

*However* - looking at `render.go`, we'd need to extend it with e.g. `directories`.

Reply:
okay, do you want me do that as part of this PR? I am kinda new to this code base, I am still trying to grok how Ignition spec works.

Reply:
No worries @gnufied! 

If the Ignition config comes through the install process Ignition will do exactly what is needed (creating the directories, dropping the file, etc..).

If it comes down through MCD then MCD itself has a subset of Ignition implementation to create files/dirs/etc.. I think this would make more sense within [this area](https://github.com/openshift/machine-config-operator/blob/635cb799b02fda6b182334e2f019fb3ae19fa01b/pkg/daemon/update.go#L271) and/or [here](https://github.com/openshift/machine-config-operator/blob/635cb799b02fda6b182334e2f019fb3ae19fa01b/pkg/daemon/update.go#L500).

Reply:
Is the goal here to just create the dir (that is always the same dir location) and never do anything else with it in the MCD?

Reply:
For clarity, templates is fine. If you need to do some kind of operations then the areas I pointed out would be where to add logic.

Reply:
Goal is just to create the directory. We don't need to do anything else in MCD.

Reply:
I agree with @cgwalters that templates makes more sense.

Reply:
Based on https://github.com/coreos/ignition/pull/706 you could probably "cheat" and make a `.dummy` file there or so (assuming kubelet won't try to load a `.dummy` or whatever file as a plugin).

Reply:
Yeah I think that would work. Kubelet seems to skip loading files that begin with `.` as plugins.

Reply:
Related: #125